client/src/AddVideoForm.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from "react";
+// import moment from "moment";
+
+function AddVideoForm({AddVideo}) {
+//   const [showValid, Setvalid] = useState(true);
+    const [newVideo, setNewVideo] = useState({});
+   const [videoTitle, setTitle] = useState("");
+  const [videoUrl, setUrl] = useState("");
+  const [videoData, setVideoData] = useState({
+    title: "",
+    url: "",
+  });
+  ;
+
+//   const handleShowValid = () => {
+//     Setvalid(!showValid);}
+  
+   function isValidUrl(url) {
+    try {
+      new URL(url);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+  const handleAddVideo = (event) => {
+    event.preventDefault();
+    videoTitle !== ""
+      ? (newVideo.title = videoTitle)
+      : alert("Please add a title for your video.");
+
+    isValidUrl(videoUrl)
+      ? (newVideo.url = videoUrl)
+      : alert("Please add a valid url.");
+    newVideo.rating = Math.floor(Math.random() * 5000) + 1;
+    
+   
+
+     AddVideo(newVideo);
+    setVideoData({
+      title: "",
+      url: "",
+     
+    });
+newVideo.id = Math.floor(Math.random() * 1000);
```

Comment:
Better to use a proper  `UUID` generator instead of relying on 1/1000 chance that two videos will get the same ID